include/Gaffer/Signals.h

```diff
@@ -85,6 +84,11 @@ class Connection
 
 };
 
+/// Calls all slots in order, returning the result from the last
+/// slot. If not slots are connected, returns a default-constructed `T`.
+template<typename T>
+struct DefaultCombiner;
```

Comment:
The general concept (and the template parameter) is Combiner rather than caller because arbitrary combination of results is possible. So I think it's right to call this DefaultCombiner - it's the combiner you'll get if you don't specify an alternative. The bindings have the separate concept of a SignalCaller and a SlotCaller as well, and those are different things, so that's another reason to avoid Caller.